src/settings-manager/metadata-manager.test.ts

```diff
@@ -1,28 +1,42 @@
 import { beforeEach, describe, expect, it, vi } from "vitest";
 
-import { MetadataEntry, MetadataManager } from "./metadata-manager";
+import {
+  DecryptCallback,
+  EncryptCallback,
+  MetadataEntry,
+  MetadataManager,
+} from "./metadata-manager";
 
 const initialEntry = { key: "a", value: "without domain" };
+const encryptedEntry = {
+  key: "encrypted",
+  value: "ddd891446804916e3b2cba1fad9f4ebf9643e1a56794e8a9",
+}; // the value is encrypted string `new value`
 
 const entryForDomainX = { key: "a__x.com", value: "domain x value" };
 const entryForDomainY = { key: "a__y.com", value: "domain y value" };
 
-const metadata = [initialEntry, entryForDomainX, entryForDomainY];
+const metadata = [initialEntry, entryForDomainX, entryForDomainY, encryptedEntry];
 
 describe("settings-manager", () => {
   describe("metadata-manager", () => {
     const fetchMock = vi.fn(async () => metadata);
     const mutateMock = vi.fn(async (md: MetadataEntry[]) => [...metadata, ...md]);
+    const manager = new MetadataManager({
+      fetchMetadata: fetchMock,
+      mutateMetadata: mutateMock,
+      encryptionKey: "key",
+    });
```

Comment:
if metadata manager doesnt force encryption, we should not require encryption key, if someone will not use encryption

So I think its better to write it this way, WDYT

```
interface SettingsManager {
set(...);
get(...)
} 

class MetadataManager implements SettingsManager {} // or PlainMetadataManager

class EncryptedMetadataManager implements SettingsManager {} 

```

I think its quite clean